dialogs/simulation/reference_density_dialog.py

```diff
@@ -97,14 +99,41 @@ def __init__(self, recoil_element: RecoilElement,
         self.formLayout.removeRow(self.widget)
 
         self.isOk = False
+        self.isCancelled = False  # Cancel-button's state
 
         self.exec_()
 
     def _state_changed(self):
         if self.userSelectionCheckBox.isChecked():
-            self.scientific_spinbox.setDisabled(False)
+            # When the reference density is set by manually and the user
+            # cancels modifications
+            if self.element_simulation.request.\
+                    manual_reference_density_checked == self.initial_state \
+                    and self.isCancelled:
+                return
+            if self.element_simulation.request.manual_reference_density_checked:
+                self.change_state(is_disabled=True, is_checked=False)
+                self.userSelectionCheckBox.setChecked(False)
+            else:
+                reply = QtWidgets.QMessageBox.critical(
+                    self, "Critical warning",
+                    "This is an unstable feature. "
+                    "It only changes the reference density of the selected "
+                    "element %s.\n"
+                    "\nDo not continue unless you are sure what you are doing!" %
+                    self.recoil_element.element.symbol,
+                    QtWidgets.QMessageBox.Ok,
+                    QtWidgets.QMessageBox.Cancel)
```

Comment:
> This is more of a warning than a critical: https://en.wikipedia.org/wiki/Syslog#Severity_level

Yes, it's a critical **warning**.

Reply:
A critical warning would be two separate [severity levels](https://doc.qt.io/qt-5/qmessagebox.html#severity-levels-and-the-icon-and-pixmap-properties) in one, which is not possible.

As we can see in the Qt documentation, *critical* is reserved for critical errors (the most severe kind). Now if we look up "critical error" online, we can see that it's [really severe](https://www.techopedia.com/definition/6667/critical-error).

> A critical error is a serious computer error that forces the program to stop and it becomes impossible for the running program, operating system or software to continue working normally. This error might cause the computer to reboot or freeze.

Meanwhile, the thing here isn't even an error (because it's expected and normal), so according to the Qt documentation, it should be *information* at most. However, I think we can go with *warning* to catch the user's attention.